new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Installer.java

```diff
@@ -80,13 +85,23 @@ private Installer params(InstallParams params) {
    * @return true if the command succeeds 
    */
   public boolean download() {
+ 
     boolean downloadSucceeded = true;
     boolean unzipSucceeded = true;
     if (params.verify()
         && new File(DOWNLOAD_DIR, params.fileName()).exists()) {
       logger.info("File {0} already exists.", params.fileName());
     } else {
+      // check and make sure DOWNLOAD_DIR exists; will create it if it is missing
       checkDirectory(DOWNLOAD_DIR);
+      
+      // we check if we get the version of the tool correctly first
+      if (params.version() == null 
+          || params.version().length() == 0
+          || params.version().equalsIgnoreCase("latest")) {
+        logger.severe("Failed to get the latest version of {0}", params.type());
```

Comment:
I don't understand what this check is ensuring.

Reply:
When the params.version() method is called the first time,  it (unexpectedly, as you indicated below) performs some additional operations to find out the release number of the "latest" by calling getRealVersionIfNeeded(). Method getRealVersionIfNeeded may return null. This checks ensure that we did get the real version, and the code returns false to indicate a failure if finding the real version somehow failed.